third_party/conditions/apis/conditions/v1alpha1/types.go

```diff
@@ -67,6 +67,11 @@ type Condition struct {
 	// +optional
 	Severity ConditionSeverity `json:"severity,omitempty"`
 
+	// Last time the condition got an update.
+	// Can be used by the system to determine if the ConditionStatus is Unknown in certain cases.
+	// +optional
+	LastHeartbeatTime metav1.Time `json:"lastHeartbeatTime,omitempty"`
```

Comment:
Hm, even upstream has no concept of this. Why do we need it? If heartbeats can be arbitrarily delayed, why is this valuable?

Reply:
@stevekuznetsov perhaps is better to create a clusterCondition struct for this use case?

Used to monitor when the kubelet (syncer in our case) stops reporting, based on:
https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L4334

Reply:
I don't think that link is what you wanted - what did you mean to link to?

Reply:
@stevekuznetsov true, sorry, this is the correct line: https://github.com/kubernetes/kubernetes/blob/f95c70080f21fe9aa6f14b72b75078d79ef68614/pkg/apis/core/types.go#L4391

Reply:
I see. I guess having a union type here of NodeCondition and Condition is ok? @ncdc you see any challenge with upstreaming this?

Reply:
So, I've just created a new struct with the missing field and embedded the Condition struct into it, so we can avoid having to upstream any changes. I guess this could work.. wdyt @ncdc ?